src/main/scala/com/cognite/sdk/scala/v1/dataModels.scala

```diff
@@ -3,81 +3,104 @@
 
 package com.cognite.sdk.scala.v1
 
+final case class SpacedItems[A](spaceExternalId: String, items: Seq[A])
+
+final case class DataModelIdentifier(
+    space: Option[String],
+    model: String
+)
+
+object DataModelIdentifier {
+  def apply(model: String): DataModelIdentifier =
+    DataModelIdentifier(None, model)
+  def apply(space: String, model: String): DataModelIdentifier =
+    DataModelIdentifier(Some(space), model)
+}
+
 final case class DataModelProperty(
-    `type`: String,
+    `type`: PropertyType,
     nullable: Boolean = true,
-    targetModelExternalId: Option[String] = None
+    targetModel: Option[DataModelIdentifier] = None
 )
 
-final case class DataModelPropertyIndex(
-    indexName: Option[String] = None,
-    fields: Option[Seq[String]] = None
-)
+final case class ContstrainedProperty(property: String)
```

Comment:
Maybe this makes sense in the DMS domain, but what is a constrained property? From the case class it seems it's just a string property (not saying this is wrong btw, just trying to understand)

Reply:
yep, it's just that way because the encoding requires
```
[
   {
      "property": "property_external_id_1"
   },
   {
      "property": "property_external_id_2"
   }
]
```

so I created that `ConstrainedProperty` as an object that only has a string `Property`... don't know if there's a better way to handle the encoding/decoding

Reply:
No, I don't think there is (as long as we're using implicit encoding this is how we do it all over the rest of the codebase too)! But perhaps it should just be called Property, then? Unless it's actually constrained by something? :thinking: